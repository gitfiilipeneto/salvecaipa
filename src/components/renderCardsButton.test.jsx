import React from 'react';
import renderer from 'react-test-renderer';
// import Link from '../Link.react';
import AddMore from './renderCardsButton'

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <AddMore>Add</AddMore>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
    let root = component.root
  // manually trigger the callback
  const button = root.findByType('button')
  button.props.onClick()

  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
 button.props.onClick()
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


