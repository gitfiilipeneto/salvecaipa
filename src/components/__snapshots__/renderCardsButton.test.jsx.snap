// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Mais uma
  </button>
</div>
`;

exports[`Link changes the class when hovered 2`] = `
<div>
  <button
    onClick={[Function]}
  >
    Mais uma
  </button>
  <div
    className="sc-bdfBwQ jQtnUZ"
  >
    <div>
      <img />
      <div
        class="dropdown"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="btn btn-secondary btn-sm dropdown-toggle dropdown-toggle-split"
          data-toggle="dropdown"
          id="dropdownMenuButton"
          type="button"
        >
          Qual Caipa?
        </button>
        <div
          aria-labelledby="dropdownMenuButton"
          class="dropdown-menu"
        >
          <p
            class="dropdown-item"
          >
            600ml
          </p>
          <p
            class="dropdown-item"
          >
            1L
          </p>
        </div>
      </div>
      <div>
        <table
          className="sc-dlfnbm bjUGZT"
        >
          <tr>
            <th>
              Sabor
            </th>
            <th>
              teste
            </th>
          </tr>
          <tr>
            <td>
              <li>
                Morango
              </li>
              <li>
                Limão
              </li>
              <li>
                Maracujá
              </li>
              <li>
                Abacaxi
              </li>
              <li>
                Vinho com Limão
              </li>
            </td>
            <div>
              <td>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    -
                  </button>
                  0
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    -
                  </button>
                  0
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    -
                  </button>
                  0
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    -
                  </button>
                  0
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    -
                  </button>
                  0
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    +
                  </button>
                </li>
              </td>
            </div>
          </tr>
        </table>
      </div>
      <div>
        <table
          className="sc-dlfnbm bjUGZT"
        >
          <tr>
            <th>
              Especiaria
            </th>
            <th>
              Teste
            </th>
          </tr>
          <tr>
            <td>
              <li>
                Hortelã
              </li>
              <li>
                Gengibre
              </li>
              <li>
                Pimenta
              </li>
            </td>
            <div>
              <td>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    -
                  </button>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    -
                  </button>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    -
                  </button>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    +
                  </button>
                </li>
              </td>
            </div>
          </tr>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Link changes the class when hovered 3`] = `
<div>
  <button
    onClick={[Function]}
  >
    Mais uma
  </button>
  <div
    className="sc-bdfBwQ jQtnUZ"
  >
    <div>
      <img />
      <div
        class="dropdown"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="btn btn-secondary btn-sm dropdown-toggle dropdown-toggle-split"
          data-toggle="dropdown"
          id="dropdownMenuButton"
          type="button"
        >
          Qual Caipa?
        </button>
        <div
          aria-labelledby="dropdownMenuButton"
          class="dropdown-menu"
        >
          <p
            class="dropdown-item"
          >
            600ml
          </p>
          <p
            class="dropdown-item"
          >
            1L
          </p>
        </div>
      </div>
      <div>
        <table
          className="sc-dlfnbm bjUGZT"
        >
          <tr>
            <th>
              Sabor
            </th>
            <th>
              teste
            </th>
          </tr>
          <tr>
            <td>
              <li>
                Morango
              </li>
              <li>
                Limão
              </li>
              <li>
                Maracujá
              </li>
              <li>
                Abacaxi
              </li>
              <li>
                Vinho com Limão
              </li>
            </td>
            <div>
              <td>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    -
                  </button>
                  0
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    -
                  </button>
                  0
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    -
                  </button>
                  0
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    -
                  </button>
                  0
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    -
                  </button>
                  0
                  <button
                    className="sc-gsTCUz dAWXAT"
                    onClick={[Function]}
                  >
                    +
                  </button>
                </li>
              </td>
            </div>
          </tr>
        </table>
      </div>
      <div>
        <table
          className="sc-dlfnbm bjUGZT"
        >
          <tr>
            <th>
              Especiaria
            </th>
            <th>
              Teste
            </th>
          </tr>
          <tr>
            <td>
              <li>
                Hortelã
              </li>
              <li>
                Gengibre
              </li>
              <li>
                Pimenta
              </li>
            </td>
            <div>
              <td>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    -
                  </button>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    -
                  </button>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    +
                  </button>
                </li>
                <li>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    -
                  </button>
                  <button
                    className="sc-gsTCUz dAWXAT"
                  >
                    +
                  </button>
                </li>
              </td>
            </div>
          </tr>
        </table>
      </div>
    </div>
  </div>
</div>
`;
